app/services/food_service.rb

```diff
@@ -0,0 +1,7 @@
+class FoodService
+  def self.food_search(food_name)
+    url = "https://api.nal.usda.gov/fdc/v1/foods/search?api_key=#{ENV['FOOD_API_KEY']}&query=#{food_name}"
+    response = Faraday.get(url)
+    json = JSON.parse(response.body, symbolize_names: true)
+  end
+end
```

Comment:
Your connection did not work, and that's where that error was coming from that you mentioned in your form. In the submission form, you wrote that the environment variable you had in your application.yml file used the variable of `FOOD_KEY`, not `FOOD_API_KEY`. So, I believe that a combination of 1. not having the api key come through with params, and 2, not having the food_name come through since you were using `params[:search]` instead of `params[:q]` is what made this error out.